ludwig/schema/hyperopt/scheduler.py

```diff
@@ -0,0 +1,72 @@
+from dataclasses import field
+from typing import Dict
+
+from marshmallow import fields, ValidationError
+from marshmallow_dataclass import dataclass
+
+from ludwig.schema import utils as schema_utils
+
+# Taken from https://github.com/ray-project/ray/blob/137a1b12c3b31a3622fa5f721a05a64e9b559b05/python/ray/tune/result.py
+DEFAULT_RESULT_KEYS = (
+    "training_iteration",
+    "time_total_s",
+    "timesteps_total",
+    "mean_accuracy",
+    "mean_loss",
+)
+TIME_TOTAL_S = "time_total_s"
+
+
+@dataclass
+class SchedulerConfig(schema_utils.BaseMarshmallowConfig):
+    """Basic scheduler settings."""
+
+    type: str = schema_utils.StringOptions(options=["async_hyperband"], default="async_hyperband", allow_none=False)
```

Comment:
@ksbrar So as far as I know, if no scheduler is provided, Ray Tune actually defaults to the FIFO scheduler type (https://docs.ray.io/en/releases-1.12.1/tune/api_docs/schedulers.html#fifoscheduler), and there are many other options as well that we support. Essentially anything that can be created by `tune.create_scheduler` (https://docs.ray.io/en/releases-1.12.1/tune/api_docs/schedulers.html#shim-instantiation-tune-create-scheduler), so I would think almost all the algorithms on the page can be supported 

cc: @ShreyaR can you confirm?

Reply:
gotcha, will add those in

Reply:
@arnavgarg1 yep, can confirm that any scheduler that tune supports should work and that the default is `FIFO`. Here's the list:
```
    "fifo": FIFOScheduler,
    "async_hyperband": AsyncHyperBandScheduler,
    "asynchyperband": AsyncHyperBandScheduler,
    "median_stopping_rule": MedianStoppingRule,
    "medianstopping": MedianStoppingRule,
    "hyperband": HyperBandScheduler,
    "hb_bohb": HyperBandForBOHB,
    "pbt": PopulationBasedTraining,
    "pbt_replay": PopulationBasedTrainingReplay,
    "pb2": _pb2_importer,
    "resource_changing": ResourceChangingScheduler,
```

Reply:
done.